docs/core/extensions/logger-message-generator.md

```diff
@@ -0,0 +1,320 @@
+    public static partial void UsingFormatSpecifier(ILogger logger, double value);
+
+    [LoggerMessage(EventId = 9, Level = LogLevel.Trace, Message = "Fixed message", EventName = "CustomEventName")]
+    public partial void LogWithCustomEventName();
+
+    [LoggerMessage(EventId = 10, Message = "Welcome to {city} {province}!")]
+    public partial void LogWithDynamicLogLevel(string city, LogLevel level, string province);
+
+    public void TestLogging()
+    {
+        LogWithCustomEventName();
+        LogWithDynamicLogLevel("Vancouver", Level = LogLevel.Warning, "BC");
+        LogWithDynamicLogLevel("Vancouver", Level = LogLevel.Information, "BC");
+        double val = 12345.6789;
+        Log.UsingFormatSpecifier(logger, val);
+    }
+}
+```
+output to `TestLogging()` using SimpleConsole:
+```
+trce: LoggingExample[9]
+      Fixed message
+warn: LoggingExample[10]
+      Welcome to Vancouver BC!
+info: LoggingExample[10]
+      Welcome to Vancouver BC!
+crit: LoggingExample[20]
+      Value is 1.234568E+004
+```
+
+same console logs formatted using JsonConsole:
+```
+{
+  "EventId": 9,
+  "LogLevel": "Trace",
+  "Category": "LoggingExample",
+  "Message": "Fixed message",
+  "State": {
+    "Message": "Fixed message",
+    "{OriginalFormat}": "Fixed message"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Warning",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 10,
+  "LogLevel": "Information",
+  "Category": "LoggingExample",
+  "Message": "Welcome to Vancouver BC!",
+  "State": {
+    "Message": "Welcome to Vancouver BC!",
+    "city": "Vancouver",
+    "province": "BC",
+    "{OriginalFormat}": "Welcome to {city} {province}!"
+  }
+}
+{
+  "EventId": 20,
+  "LogLevel": "Critical",
+  "Category": "LoggingExample",
+  "Message": "Value is 1.234568E+004",
+  "State": {
+    "Message": "Value is 1.234568E+004",
+    "value": 12345.6789,
+    "{OriginalFormat}": "Value is {value:E}"
+  }
+}
+```
+
+### Benefits with the source generator approach:
+
+- Allows the logging structure to be preserved and enables the exact format syntax required by https://messagetemplates.org/
+- Allows supplying alternative names for the holes and using format specifiers.
+- Allows to pass all of original data as-is without any complication around how it's stored prior to something being done with it other than creating a string.
+- Provides logging-specific diagnostics, e.g. it emits warnings for duplicate event ids.
+
+#### Benefits compared to using LoggerMessage.Define directly
+
+- Shorter and simpler syntax than current approach
+- Guided developer experience - the generator gives warnings to help developers do the right thing
+- Support for an arbitrary # of logging parameters. current approach tops out at 6
```

Comment:
```suggestion
- Support for an arbitrary # of logging parameters. LoggerMessage.Define tops out at 6
```
? "Current appraoch" can also be `ILogger.Log`, though I see the header about LoggerMessage.Define, but be explicit.

Other bullets in this section too.